src/components/shared/landscape-view-manager/landscape-view-manager-component.jsx

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from 'react';
+
+const LandscapeViewManager = ({ view, zoomLevelTrigger, onZoomChange, query, isLandscapeMode }) => {
+  const [ isInteracting, setInteracting ] = useState(false);
+  const [ isUpdating, setUpdating ] = useState(false);
+  const [ landscapeView, setLandscapeView ] = useState(false);
+
+  const isLandscapeViewOnEvent = (zoomValue, landscapeView) => zoomValue >= zoomLevelTrigger && landscapeView === false;
+  const isLandscapeViewOffEvent = (zoomValue, landscapeView) => zoomValue < zoomLevelTrigger && landscapeView === true;
```

Comment:
may be simplified to `isLandscapeViewOffEvent  = (zoomValue, landscapeView) =>  !isLandscapeViewOnEvent(zoomValue, landscapeView)`

Reply:
also, maybe we can move these two functions to `selectors` file and pass it through props?

Reply:
The proposed change to refactor `isLandscapeViewOffEvent` will not work. Not in all circumstances that `isLandscapeViewOnEvent` is `false` `isLandscapeViewOffEvent` is `true` (to trigger the change both, zoom and URL param, conditions should match)